data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/buffer/BlockingBufferTests.java

```diff
@@ -62,14 +64,35 @@ public void testInsertNull() throws TimeoutException {
     }
 
     @Test(expected = TimeoutException.class)
-    public void testNoEmptySpace() throws TimeoutException {
+    public void testNoEmptySpaceWriteOnly() throws TimeoutException {
         final BlockingBuffer<Record<String>> blockingBuffer = new BlockingBuffer<>(1, TEST_BATCH_SIZE,
                 TEST_PIPELINE_NAME);
         assertThat(blockingBuffer, notNullValue());
         blockingBuffer.write(new Record<>("FILL_THE_BUFFER"), TEST_WRITE_TIMEOUT);
         blockingBuffer.write(new Record<>("TIMEOUT"), TEST_WRITE_TIMEOUT);
     }
 
+    @Test
```

Comment:
These tests should have comments describing what they do, because they are not exactly readable. One possible format is the if-when-then format:

//Describe setup (If)

//Describe test actions (when)

//Describe assertions (then)